lib/wrappers/transform-node.js

```diff
@@ -21,20 +60,44 @@ module.exports = class TransformNodeWrapper extends NodeWrapper {
     return new Promise(resolve => {
       startTime = process.hrtime();
 
+      if (this.memoizationEnabled(this.nodeInfo.volatile)) {
+        let shouldBuild = this.shouldBuild();
+
+        if (!shouldBuild) {
+          this.buildState.built = false;
+          return resolve(); // Noop the build since inputs did not change
+        }
+      }
+
       if (!this.nodeInfo.persistentOutput) {
         rimraf.sync(this.outputPath);
         fs.mkdirSync(this.outputPath);
       }
 
       resolve(this.callbackObject.build());
+
+      // If the plugin is volatile then it will be responsible for calling revise.
+      if (!this.nodeInfo.volatile) {
```

Comment:
even if the node is volatile, that only dictates that we DO build it, we should always revise.
Conditional/manual revise tracking should be future work.